saleor/tests/e2e/checkout/discounts/test_checkout_with_promotion_and_voucher.py

```diff
@@ -260,6 +271,10 @@ def test_checkout_with_promotion_and_voucher_CORE_2107(
     assert order_line["undiscountedUnitPrice"]["gross"]["amount"] == float(
         product_variant_price
     )
-    assert order_line["unitDiscountReason"] == (
-        f"Entire order voucher code: {voucher_code} & Promotion: {promotion_id}"
-    )
+    assert f"Promotion: {promotion_id}" in order_line["unitDiscountReason"]
+
+    if legacy_voucher_propagation:
```

Comment:
suggestion (non-blocking): IMO it's should be separate test. It will be much easier to remove a separate test when we remove the legacy flow.